### DISCUSSION TITLE
Home Assistant OS 10.0.rc2

### DISCUSSION THREAD

**Discussion by agners** [home-assistant/operating-system#2432] [Announcements]
## Changes

* updated generic_raw_uart and rpi-rf-mod to latest versions. (#2431) @jens-maus

## Home Assistant Operating System

* Fix swapfile creation for all memory sizes (#2427) @agners

<hr /><em>This discussion was created from the release <a href='https://github.com/home-assistant/operating-system/releases/tag/10.0.rc2'>Home Assistant OS 10.0.rc2</a>.</em>

**Comment by wastez** [comment]
And how can I change to v10. 
Dev is already v11 (ok now there is no difference but soon)

And Beta Channel is still v9.5. 

Would be great if v10 could be pushed to Beta Channel.

**Reply by tjorim** [reply]
Give it some time to build and release: https://github.com/home-assistant/operating-system/actions/runs/4535351690

**Reply by tjorim** [reply]
Edit, released now: https://github.com/home-assistant/version/commit/58d78cc7a7c2b8d4441c2e112b656cd4eacca368

**Reply by agners** [reply]
We run into some issues with the new zswap memory architecture it seems. Some add-ons don't start with out of memory errors (e.g. Node RED). I'll have to pull 10.0.rc2 as well it seems :cold_sweat:

**Reply by tjorim** [reply]
Working nicely for my add-ons here, not using Node-RED at the moment but can indeed see the issue with it.

**Reply by wastez** [reply]
Until now everything seems to run fine here too but i'm not using node-red.
Strange that zswap issues such a problem but i don't know what you should do (if you want to keep zswap), seems the provider of the add-on has to update it.
Do you know about more add-ons which has this problem or is it just Node-Red?

**Reply by agners** [reply]
There is also the NUT addon which seems to cause issues, see https://github.com/home-assistant/operating-system/issues/2433.

Meanwhile, Node RED add-on is fixed.

The issues seem not to be zswap related, but related to the new Linux kernel 6.1 LTS. It seems that it's overcommitting heuristic behaves a bit different than previously, leading more often to out of memory errors than before.

**Reply by yousaf465** [reply]
HAOS not starting after update to v10, might need to restore backup.
it remains stuck on this screen 
memory and swap usage 96%? 
![image](https://user-images.githubusercontent.com/83491212/228393388-eeae4611-d9fe-4dfc-bdbd-717b19e75a72.png)

![image](https://user-images.githubusercontent.com/83491212/228393836-0a670d35-0f65-4a88-b8e7-2c4d77da37f2.png)

**Reply by agners** [reply]
@yousaf465 what was your Memory use before HAOS 10.0.rc2? What board/system are you running on?

You can revert back to HAOS 9.5 using `ha os update --version 9.5` in the terminal or console.

**Reply by yousaf465** [reply]
> @yousaf465 what was your Memory use before HAOS 10.0.rc2? What board/system are you running on?
> 
> You can revert back to HAOS 9.5 using `ha os update --version 9.5` in the terminal or console.

memory use was 76% average with swap around 30%. 
it is RPI4 2GB

**Reply by agners** [reply]
Hm, that is quite a bit more.

Can you downgrade again and `ha os update --version 9.5`, check the values, upgrade again and see if it really is just the OS update which is changing this?

What would be interesting is the memory usage of the individual process to better understand which process is causing this higher memory load. To print the memory usage of all processes in the system you'd need SSH access to the underlying OS (port 22222, see [this instructions how-to setup](https://developers.home-assistant.io/docs/operating-system/debugging#ssh-access-to-the-host)) and the output of this command:

```
ps -o pid,user,%mem,size,command ax | sort -b -k3 -r
```

**Reply by agners** [reply]
@yousaf465 what add-ons are you using?

**Reply by yousaf465** [reply]
I am using multiple add-ons, I am actually on job right now so don't have access to the rpi4 right now.
I did try installing new installation and updated the OS to 10.0.rc2, (without installing any add-ons) , faced same issue. HA become inaccessible. 
I will definitely post memory usage for each processes, as soon as I get back home.

**Reply by yousaf465** [reply]
I couldn't SSH into even after generating the authorization key via putty and putting it in on USB. instructions are little bit vague and disorganized 
I can confirm this happens with Home Assistant OS 10.0.rc2 
![image](https://user-images.githubusercontent.com/83491212/228748396-b5fbee94-c59e-4d5f-b0ec-87c8d760f95d.png)
half hour after updating to 10.0.rc2
![image](https://user-images.githubusercontent.com/83491212/228748552-dfc4fef6-3d47-4a87-b873-aeb6d7b82c91.png)

**Reply by yousaf465** [reply]
and trying to downgrade 
```ha os update --version 9.5
Processing... Done.

Error: 'OSManager.update' blocked from execution, system is not running - CoreState.STARTUP

```

**Reply by agners** [reply]
Does that still happen after waiting for a while?

```
ha supervisor logs
```

Should give you hints why it is stuck in startup. You'll have first to figure out why Supervisor is causing problems.

In any case, what add-ons are you using? Can you stop/disable them, as well as custom integrations?

**Reply by yousaf465** [reply]
> Does that still happen after waiting for a while?
> 
> ```
> ha supervisor logs
> ```
> 
> Should give you hints why it is stuck in startup. You'll have first to figure out why Supervisor is causing problems.
> 
> In any case, what add-ons are you using? Can you stop/disable them, as well as custom integrations?
although all seems have been stopped, but that isn't the case, e.g ABDSB is running fine but start up page says it's stopped. 
![image](https://user-images.githubusercontent.com/83491212/228892496-2aa89aac-35e8-40a7-a958-85f294b3993c.png)
![image](https://user-images.githubusercontent.com/83491212/228894180-833bf746-e805-4aa5-b60f-d30d113ab71f.png)

**Reply by agners** [reply]
"Check Home Assistant configuration" you can uninstall, that no longer exists.

Make sure EMQX is the latest version (released yesterday), it would allocate lots of memory on 10.0rc2! Maybe that helps getting the issue resolved.

If not, with so many add-on it is very hard to tell what is going on. Definitely try to downgrade to HAOS 9.5 and see if that really is stable.

**Reply by yousaf465** [reply]
why you need root access when you have got glances 
![image](https://user-images.githubusercontent.com/83491212/228950986-40267007-016b-4593-a6cb-ced9bb4e72fe.png)

**Reply by yousaf465** [reply]
![image](https://user-images.githubusercontent.com/83491212/232733561-1c094830-5b23-4759-b146-08e52e997b73.png)
same issue with 10.0.

**Reply by agners** [reply]
What is the typical memory load on your system? See Settings > System > Hardware. You should not go beyond 80%. You can also use the system console to check:

```
ha > login
# free -h
```

**Reply by yousaf465** [reply]
> What is the typical memory load on your system? See Settings > System > Hardware. You should not go beyond 80%. You can also use the system console to check:
> 
> ```
> ha > login
> # free -h
> ```

usually it is 76%, but this 10.0, initially it increases to 96%, which is reduced to 56% on it's own after HA bootup and at least 3 hours have passed. during which many add-ons are closed automatically such as EQMX.

**Reply by yousaf465** [reply]
> What is the typical memory load on your system? See Settings > System > Hardware. You should not go beyond 80%. You can also use the system console to check:
> 
> ```
> ha > login
> # free -h
> ```

```➜  ~ free -h
              total        used        free      shared  buff/cache   available
Mem:           1.8G        1.7G       47.0M           0       84.5M       61.4M
Swap:          4.0G        2.8G        1.2G
➜  ~
```
with 10.0
